WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift

```diff
@@ -65,21 +65,24 @@ final class HubMenuViewModel: ObservableObject {
 
     private var cancellables: Set<AnyCancellable> = []
 
+    let tapToPayBadgePromotionChecker: TapToPayBadgePromotionChecker
+
     init(siteID: Int64,
          navigationController: UINavigationController? = nil,
+         tapToPayBadgePromotionChecker: TapToPayBadgePromotionChecker,
          featureFlagService: FeatureFlagService = ServiceLocator.featureFlagService,
          stores: StoresManager = ServiceLocator.stores,
          generalAppSettings: GeneralAppSettingsStorage = ServiceLocator.generalAppSettings) {
         self.siteID = siteID
         self.navigationController = navigationController
+        self.tapToPayBadgePromotionChecker = tapToPayBadgePromotionChecker
         self.stores = stores
         self.featureFlagService = featureFlagService
         self.generalAppSettings = generalAppSettings
         self.switchStoreEnabled = stores.isAuthenticatedWithoutWPCom == false
         observeSiteForUIUpdates()
         observePlanName()
-        listenToNewFeatureBadgeReloadRequired()
-        retrieveShouldShowNewFeatureBadgeOnPaymentsValue()
+        tapToPayBadgePromotionChecker.$shouldShowTapToPayBadges.share().assign(to: &$shouldShowNewFeatureBadgeOnPayments)
```

Comment:
💯 

Should we wrap this line in a method? Sort of "bind/observe badge changes"?

Reply:
Yeah, it is a bit wordy. I'll add this to a list to do when I have a little time.